django/template/base.py

```diff
@@ -535,7 +535,8 @@ def extend_nodelist(self, nodelist, node, token):
         if node.must_be_first and nodelist.contains_nontext:
             raise self.error(
                 token,
-                "%r must be the first tag in the template." % node,
+                '{%% %s %%} must be the first tag in "%s".'
+                % (token.contents, self.origin.template_name),
             )
```

Comment:
We can also render a template from a string, in which case this error is:
`{% extends 'base.html' %} must be the first tag in "None".`

I suggest we make an update to handle this:
```python
            if self.origin.template_name:
                origin = f'"{self.origin.template_name}"'
            else:
                origin = "the template"
            raise self.error(
                token,
                '{%% %s %%} must be the first tag in %s.'
                % (token.contents, origin),
           )
```

Reply:
Also, in the first branch, let's use:

```
origin = repr(self.origin.template_name)
```

That will use single quotes and will handle escaping any quotes that happen to appear in the template name, like: `'the \'quoted\' template.html'`.